static/app/components/charts/sunburstChart.tsx

```diff
@@ -0,0 +1,288 @@
+import 'echarts/lib/component/tooltip';
+import 'echarts/lib/component/grid';
+import 'echarts/lib/component/graphic';
+import 'echarts/lib/component/toolbox';
+import 'zrender/lib/svg/svg';
+
+import {useState} from 'react';
+import {useTheme} from '@emotion/react';
+import styled from '@emotion/styled';
+import type {SunburstSeriesOption, TooltipComponentOption} from 'echarts';
+import * as echarts from 'echarts/core';
+import ReactEchartsCore from 'echarts-for-react/lib/core';
+
+import {getTooltipStyles} from 'sentry/components/charts/baseChart';
+import {CHART_TOOLTIP_VIEWPORT_OFFSET} from 'sentry/components/charts/components/tooltip';
+import type {EChartClickHandler, EChartMouseOverHandler} from 'sentry/types/echarts';
+
+function bfsFilter(root: ProcessedSunburstData, maxDepth: number) {
+  const editableTree = structuredClone(root);
+  const queue = [editableTree];
+
+  while (queue.length > 0) {
+    const node = queue.shift();
+
+    if (!node) {
+      continue;
+    }
+
+    if (node.depth === maxDepth) {
+      node.children = [];
+      node.value = 1;
+      continue;
+    }
+
+    node?.children?.forEach(child => {
+      queue.push(child);
+    });
+  }
+
+  return editableTree;
+}
+
+export interface SunburstData {
+  children: SunburstData[];
+  coverage: number;
+  fullPath: string;
+  name: string;
+  value: number;
+  dir?: boolean;
+}
+
+interface ProcessedSunburstData extends SunburstData {
+  depth?: number;
+  itemStyle?: SunburstSeriesOption['itemStyle'];
+  type?: 'dir' | 'file';
```

Comment:
should this be optional if it's only these two or can we handle undefined?

Reply:
These values are optional because they are set after the data has been passed so they're required to be optional.